design/organization.go

```diff
@@ -0,0 +1,77 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var _ = a.Resource("organization", func() {
+
+	a.BasePath("/organizations")
+
+	a.Action("create", func() {
+		a.Routing(
+			a.POST(""),
+		)
+		a.Description("Create a new organization")
+		a.Payload(CreateOrganizationRequestMedia)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.Created, CreateOrganizationResponseMedia)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.BadRequest, JSONAPIErrors)
+	})
+
+	a.Action("list", func() {
+		a.Routing(
+			a.GET(""),
+		)
+		a.Description("Lists organizations that the user has access to")
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.OK, organizationArray)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.BadRequest, JSONAPIErrors)
+	})
+})
+
+var CreateOrganizationRequestMedia = a.MediaType("application/vnd.create_organization_request+json", func() {
+	a.Description("Request payload required to create a new organization")
+	a.Attributes(func() {
+		a.Attribute("name", d.String, "The name of the new organization")
+	})
+	a.View("default", func() {
+		a.Attribute("name")
+	})
+})
+
+var CreateOrganizationResponseMedia = a.MediaType("application/vnd.create_organization_response+json", func() {
+	a.Description("Response returned when creating a new organization")
+	a.Attributes(func() {
+		a.Attribute("organization_id", d.String, "The identifier of the new organization")
+	})
+	a.View("default", func() {
+		a.Attribute("organization_id")
+	})
+})
+
+var organizationArray = a.MediaType("application/vnd.organization-array+json", func() {
+	a.UseTrait("jsonapi-media-type")
+	a.TypeName("OrganizationArray")
+	a.Description("Organization Array")
+	a.Attributes(func() {
+		a.Attribute("organizations", a.ArrayOf(organizationData))
```

Comment:
Let's call this attribute `data` to keep some consistency with other APIs. In the future, we might have to consider pagination. In the jsonapi spec, we would usually have a `type = organizations` if at all. Let's skip the latter and use `data` now?

Reply:
Done.